<launch>
        <rosparam command="load" file="$(find spirit_utils)/../config/project_params.yaml"/>

        <!-- Launch the node to generate simple terrain -->
        <node pkg="spirit_utils" type="terrain_map_publisher_node" name="terrain_map_publisher" output="screen" />

        <!-- Load the image of the terrain map -->
        <node pkg="grid_map_demos" type="image_publisher.py" name="image_publisher" output="screen">
            <param name="image_path" value="$(find spirit_utils)/data/slope.png" />
            <param name="topic" value="~image" />
        </node>
                
        <!-- Launch the node to publish the global body plan -->
        <node name="global_body_planner" pkg="global_body_planner" type="global_body_planner_node" output="screen"/>

        <!-- Launch the node to publish the local footstep plan -->
        <node name="local_footstep_planner" pkg="local_footstep_planner" type="local_footstep_planner_node" output="screen"/>

        <!-- Launch the node the publish state estimate messages-->
        <node name="ekf_estimator" pkg="ekf_estimator" type="ekf_estimator_node" output="screen"/>

        <!-- Launch the node to publish the mpc controller -->
        <node name="mpc_controller" pkg="mpc_controller" type="mpc_controller_node" output="screen"/>

        <!-- Launch the stand-in robot interface node -->
        <node name="robot_interface" pkg="spirit_utils" type="robot_interface_node" output="screen"/>

        <!-- Launch the grid map visualizer -->
        <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen">
          <!-- <rosparam command="load" file="$(find global_body_planner)/config/simple_terrain.yaml" /> -->
        </node>

        <!-- Point the robot description to this URDF (which has meshes attached) -->
        <param name="robot_description" command="cat $(find spirit_description)/urdf/spirit_meshes.urdf" />

        <!-- Launch the nodes to publish all the robot tfs and visualize in rviz -->
        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

        <!-- Launch the node to publish the visual topics for the plans -->
        <node name="rviz_interface" pkg="spirit_utils" type="rviz_interface_node" output="screen"/>

        <!-- Launch the nodes to publish all the robot tfs and visualize in rviz -->
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find spirit_utils)/rviz/example_with_terrain.rviz"/>


</launch>